File: solutions_5658571765186560_1/C++/PAP/D2.cpp
#include<iostream>

using namespace std;

int main()
{
	int t;
	cin >> t;
	for(int tt = 1; tt <= t; tt++)
	{
		int r, c, x;
		cin >> x >> r >> c;
		if(r < c)
			swap(r, c);
		cout << "Case #" << tt << ": ";
		if(x >= 7)
			cout << "RICHARD" << endl;
		else if(x == 1)
			cout << "GABRIEL" << endl;
		else if(x == 2)
		{
			if(r * c % 2 == 0)
				cout << "GABRIEL" << endl;
			else
				cout << "RICHARD" << endl;
		}
		else if(x == 3)
		{
			if(r < 3 || c < 2 || r * c % 3 != 0)
				cout << "RICHARD" << endl;
			else
				cout << "GABRIEL" << endl;
		}
		else if(x == 4)
		{
			if(r < 4 || c < 3 || r * c % 4 != 0)
				cout << "RICHARD" << endl;
			else
				cout << "GABRIEL" << endl;
		}
		else if(x == 5)
		{
			if(r < 5 || c < 3 || r * c % 5 != 0)
				cout << "RICHARD" << endl;
			else
				cout << "GABRIEL" << endl;
		}
		else
		{
			if(r < 6 || c < 4 || r * c % 6 != 0)
				cout << "RICHARD" << endl;
			else
				cout << "GABRIEL" << endl;
		}
	}
	return 0;
}

File: solutions_5658571765186560_0/Python/awesomepenguin/D2.py
t = int(raw_input())

for qq in xrange(1, t+1):
    x, r, c = map(int, raw_input().split())
    if r > c: (r, c) = (c, r)
    n = r * c

    if x == 1: winner = 'GABRIEL'
    elif x == 2:
        if n % 2 == 0: winner = 'GABRIEL'
        else: winner = 'RICHARD'
    elif x == 3:
        if r == 1 and c == 1: winner = 'RICHARD'
        elif r == 1 and c == 2: winner = 'RICHARD'
        elif r == 2 and c == 2: winner = 'RICHARD'
        elif r == 1 and c == 3: winner = 'RICHARD'
        elif r == 1 and c == 4: winner = 'RICHARD'
        elif r == 2 and c == 4: winner = 'RICHARD'
        elif r == 4 and c == 4: winner = 'RICHARD'
        else: winner = 'GABRIEL'
    else: #x = 4
        if r == 1 and c == 1: winner = 'RICHARD'
        elif r == 1 and c == 2: winner = 'RICHARD'
        elif r == 2 and c == 2: winner = 'RICHARD'
        elif r == 1 and c == 3: winner = 'RICHARD'
        elif r == 2 and c == 3: winner = 'RICHARD'
        elif r == 3 and c == 3: winner = 'RICHARD'
        elif r == 1 and c == 4: winner = 'RICHARD'
        elif r == 2 and c == 4: winner = 'RICHARD'
        else: winner = 'GABRIEL'
    
    print 'Case #%d: %s' % (qq, winner)
